shell/platform/android/android_context_gl.cc

```diff
@@ -234,41 +229,29 @@ bool AndroidContextGL::ClearCurrent() {
   return true;
 }
 
-bool AndroidContextGL::SwapBuffers() {
+bool AndroidContextGL::SwapBuffers(EGLSurface surface) {
   TRACE_EVENT0("flutter", "AndroidContextGL::SwapBuffers");
-  return eglSwapBuffers(environment_->Display(), surface_);
+  return eglSwapBuffers(environment_->Display(), surface);
 }
 
-SkISize AndroidContextGL::GetSize() {
+SkISize AndroidContextGL::GetSize(EGLSurface surface) {
   EGLint width = 0;
   EGLint height = 0;
 
-  if (!eglQuerySurface(environment_->Display(), surface_, EGL_WIDTH, &width) ||
-      !eglQuerySurface(environment_->Display(), surface_, EGL_HEIGHT,
-                       &height)) {
+  if (!eglQuerySurface(environment_->Display(), surface, EGL_WIDTH, &width) ||
```

Comment:
If you are going to make EGLSurface an RAII object, it may make sense to put these behind accessors. Just a thought.

Reply:
I tried to use ` fml/unique_object.h`, but I realized it was a bit tricky for one reason:
To destroy an `EGLSurface`, an instance of EGLDisplay is required. This is currently held by `AndroidEnvironmentGL`.  While we currently just use `EGL_DEFAULT_DISPLAY`, I shouldn't assume that in a `Free(EGLSurface* surface)`  method.

Reply:
Discussed this via chat.